### PR TITLE
Signing only required in CI when taskGraph includes publishingPlugins

### LINKED ISSUE #478: Signing configuration is not compatible with Gradle 8.1
Currently, the project's signing configuration is not compatible with Gradle 8.1. 

We should update the build in a way to make it compatible with Gradle 8.1.

### PR DESCRIPTION
**Pull request by cdsap**
Closes #478 

#### Gradle 8.0.2
- [x]  CI passing https://github.com/gradle/android-cache-fix-gradle-plugin/actions/runs/4503448211 and https://github.com/gradle/android-cache-fix-gradle-plugin/actions/runs/4503449449
- [x] Local publish to maven local: https://ge.solutions-team.gradle.com/s/4wb4vqsnvsjd4
- [x] Local, project using the published version [https://ge.solutions-team.gradle.com/s/ejdoyjlfglxxi/build-dependencies?focusedDependency=W[…]bMCwwLFs3OV1dXQ&toggled=W1swXSxbMCwwXSxbMCwwLFs3OV1dXQ](https://ge.solutions-team.gradle.com/s/ejdoyjlfglxxi/build-dependencies?focusedDependency=WzAsMCw3OSxbMCwwLFs3OV1dXQ&toggled=W1swXSxbMCwwXSxbMCwwLFs3OV1dXQ)
- [x] Published plugin in staging  https://github.com/gradle/ge-solutions/actions/runs/4506089290/jobs/7932503309
 
#### Gradle 8.1
- [x] CI passing: https://github.com/gradle/android-cache-fix-gradle-plugin/actions/runs/4504797075 and https://github.com/gradle/android-cache-fix-gradle-plugin/actions/runs/4504797073
- [x] Local publish to maven local: https://ge.solutions-team.gradle.com/s/uq3yh77pjzspo
- [x] Local, project using the published version: https://ge.solutions-team.gradle.com/s/afasttfzu35qi/build-dependencies?focusedDependency=WzAsMCw4MixbMCwwLFs4Ml1dXQ&toggled=W1swXSxbMCwwXV0
- [x] Published plugin in staging: https://github.com/gradle/ge-solutions/actions/runs/4505505528

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -154,8 +154,8 @@ tasks.withType(ValidatePlugins).configureEach {
 }
 
 signing {
-    // Require publications to be signed on CI. Otherwise, publication will be signed only if keys are provided.
-    required isCI
+    // Require publications to be signed on CI and publishing task included in the TaskGraph
+    required { isCI && gradle.taskGraph.hasTask(":publishPlugins") }
```
Comment (erichaagdev) [inline build.gradle:158]: IMO we can remove `isCI`. Regardless of from where the plugin is being published, signing should be required.

**Review by runningcode — COMMENTED** [review]
This looks good, but is this what the BT team recommends for this situation?

**Comment by cdsap** [comment]
@runningcode 
> This looks good, but is this what the BT team recommends for this situation?

the configuration of conditional signing depending if a task is included in the graph was based on the BT doc https://docs.gradle.org/8.1-rc-1/userguide/signing_plugin.html#sec:conditional_signing

**Comment by runningcode** [comment]
Sounds good, Please convert this draft to a PR when it is ready for review.

File: build.gradle (hunk 1 above)
Comment (cdsap) [reply build.gradle]: publishing locally the plugin shouldn't require signing. Good news is we are not using the `publishPlugins` for local  builds and `publishToMavenLocal` doesn't include the task required to sign. 

done https://github.com/gradle/android-cache-fix-gradle-plugin/pull/479/commits/f15a48bdbedfcfca982f65719f5e72d6ba5c0130

**Review by erichaagdev — APPROVED** [review]

**Pull request merged**